eng/cake/dotnet.cake

```diff
@@ -156,7 +156,11 @@ Task("dotnet-templates")
                 var projectName = template.Key.Split(":")[0];
                 var templateName = template.Key.Split(":")[1];
 
-                var framework = string.IsNullOrWhiteSpace(TestTFM) ? "" : $"--framework {TestTFM}";
+                var framework = string.IsNullOrWhiteSpace(TestTFM)
+                    ? ""
+                    : TestTFM.StartsWith("net6")
+                        ? $"--Framework {TestTFM}" // this is just needed until we ship the next MAUI SR2
+                        : $"--framework {TestTFM}";
```

Comment:
This is needed for now in CI due to the fact that the .NET SDK changed casing or changed casing strictness. The correct workaround is to make the templates support both cases using the dotnetcli.host.json. However, since we pull from NuGet, we need to wait for that fix to reach NuGet.

The fix was in: https://github.com/dotnet/maui/pull/8282

It should be going out with Maui SR 2